sklearn/datasets/openml.py

```diff
@@ -216,6 +216,53 @@ def _convert_arff_data(arff_data, col_slice_x, col_slice_y):
         raise ValueError('Unexpected Data Type obtained from arff.')
 
 
+def _convert_arff_data_dataframe(arff_data, col_slice_x, col_slice_y,
+                                 data_columns, target_column, features_dict,
+                                 nominal_attributes):
+    """
+    """
+    try:
+        import pandas as pd
+    except ImportError:
+        raise ImportError("return_frame=True requires pandas")
+
+    nominal_attributes = dict(nominal_attributes)
+
+    data = pd.DataFrame(arff_data)
+
+    # process X data
+    X = data.iloc[:, col_slice_x]
+    X.columns = data_columns
+
+    all_numeric = all(features_dict[feature]['data_type'] == 'numeric'
+                      for feature in data_columns)
+
+    if all_numeric:
+        X = X.astype(np.float64, copy=False)
+    else:
+        X_cols = []
+
+        for feature in data_columns:
+            data_type = features_dict[feature]['data_type']
+            if data_type == 'numeric':
+                X_cols.append(X[feature].astype(np.float64))
+            elif data_type == 'nominal':
+                categories = nominal_attributes[feature]
+                X_cols.append(pd.Series(
+                    pd.Categorical(X[feature], categories=categories),
```

Comment:
Categories are stored as strings. It could be useful to try converting them to integer if they were orginally stored in this way?

Reply:
This is probably in the scope of OpenML python instead.